test/test_dynamic_port_binding.rb

```diff
@@ -382,6 +382,45 @@ module Syskit
             @task = syskit_stub_deploy_configure_and_start(@task_m)
         end
 
+        describe "policy" do
+            attr_reader :task, :port_binding
+
+            before do
+                @port_binding = flexmock
+                @accessor = DynamicPortBinding::Accessor.new(@port_binding)
+                flexmock(@accessor)
+                    .should_receive(:create_accessor)
+                    .explicitly
+                    .with(@task.out_port).and_return { @task.out_port.reader }
+            end
+
+            it "expects no policy if init_policy is not called" do
+                flexmock(@task.out_port)
+                    .should_receive(:reader)
+                    .and_return({})
+
+                @accessor.create_accessor(@task.out_port)
+            end
+
+            it "expects init: true policy if init_policy(true) is called" do
+                @task.out_port.model.init_policy(true)
+                flexmock(@task.out_port)
+                    .should_receive(:reader)
+                    .and_return({ init: true })
```

Comment:
Should be `with` and not `and_return`. Which is why the tests pass even though the implementation is wrong.